pandas/io/sql.py

```diff
@@ -108,13 +108,18 @@ def _parse_date_columns(data_frame, parse_dates):
     # we could in theory do a 'nice' conversion from a FixedOffset tz
     # GH11216
     for col_name, df_col in data_frame.items():
-        if is_datetime64tz_dtype(df_col) or col_name in parse_dates:
+        # use updated infer_dtype instead of is_datetime64tz_dtype
+        # this will handle datetime Series with different offsets
+        # see GH30207
+        if lib.infer_dtype(df_col) == "datetimetz":
+            data_frame[col_name] = to_datetime(df_col, utc=True)
```

Comment:
As mentioned before just use `_handle_date_column(df_col, utc=lib.infer_dtype(df_col) == "datetimetz", format=fmt)` or something similar.

Reply:
Oh ok I did not get this before. So since [is_datetime64tz_dtype](https://github.com/pandas-dev/pandas/blob/v0.25.3/pandas/core/dtypes/common.py#L539-L574) is used to recognize the extension dtype [DatetimeTZDtype](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DatetimeTZDtype.html) specifically and since a Series with mixed timezones/offsets will result in a dtype "object" I would propose to modify the function as such (to me this seems close to what you wrote):

```
def _parse_date_columns(data_frame, parse_dates):
    """
    Force non-datetime columns to be read as such.
    Supports both string formatted and integer timestamp columns.
    """
    parse_dates = _process_parse_dates_argument(parse_dates)

    # we want to coerce datetime64_tz dtypes for now to UTC
    # we could in theory do a 'nice' conversion from a FixedOffset tz
    # GH11216
    for col_name, df_col in data_frame.items():
        # infer_dtype == "datetimetz" will work with different timezones/offsets
        # (in such cases the dtype is object and not DatetimeTZDtype)
        is_datetimetz = (
            is_datetime64tz_dtype(df_col) or lib.infer_dtype(df_col) == "datetimetz"
        )
        if is_datetimetz or col_name in parse_dates:
            try:
                fmt = parse_dates[col_name]
            except TypeError:
                fmt = None
            data_frame[col_name] = _handle_date_column(
                df_col, utc=is_datetimetz, format=fmt
            )
    return data_frame
```

Reply:
As @jreback mentionned let's do this in another PR :+1: